lib/player/db/postgres.go

```diff
@@ -0,0 +1,138 @@
+// Teleport
+// Copyright (C) 2024 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package db
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/gravitational/teleport/api/types/events"
+)
+
+// PostgresTranslator is responsible for converting PostgreSQL session recording
+// events into their text representation.
+type PostgresTranslator struct {
+	sessionStartAt                 time.Time
+	preparedStatements             map[string]string
+	lastPreparedStatement          string
+	lastPreparedStatementArguments []string
+	expectingResult                bool
+}
+
+// NewPostgresTranslator constructs a new instance of PostgreSQL translator.
+func NewPostgresTranslator() *PostgresTranslator {
+	return &PostgresTranslator{
+		preparedStatements: make(map[string]string),
+		expectingResult:    false,
+	}
+}
+
+// TranslateEvent converts PostgreSQL audit events into session print events.
+func (p *PostgresTranslator) TranslateEvent(evt events.AuditEvent) *events.SessionPrint {
+	switch e := evt.(type) {
+	case *events.DatabaseSessionQuery:
+		p.expectingResult = true
+		return p.generatePrintEvent(e.Metadata, p.generateCommandPrint(e.DatabaseMetadata, e.DatabaseQuery))
+	case *events.DatabaseSessionCommandResult:
+		if p.expectingResult {
```

Comment:
Is `p.expectingResult` necessary?

Reply:
It is used to skip the results of commands that were also skipped. It is currently being used by skipped prepared statement executions and function calls.